File: scala/src/main/scala/codewars/magnetParticules.scala
package codewars


object Magnets {
  def doubles(maxk: Int, maxn: Int): Double = {
    val pairs = for{
      k <- 1 to maxk;
      n <- 1 to maxn
    } yield v(k,n)

    pairs.toList.foldLeft(0d)(_+_)
  }

  def v(k: Int, n: Int): Double = {
    1 / (k*Math.pow((n+1).toDouble, (2*k).toDouble))
  }

  def main(args: Array[String]) = {
    println(doubles(1,3))
  }
}

File: go/pkg/magnetParticules/magnetParticules.go
package magnetParticules

import (
	. "math"
)

func v(k float64, n float64) float64{
	return 1 /(k*Pow((n+1), 2*k))
}

func Doubles(maxk, maxn int) float64 {
	var sum float64 = 0
    for k:=1; k<=maxk; k++ {
		for n:=1; n<=maxn; n++ {
			sum += v(float64(k),float64(n))
		}
	}
	
	return sum
}
